localization/pose_instability_detector/src/pose_instability_detector.cpp

```diff
@@ -0,0 +1,176 @@
+// Copyright 2023- Autoware Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "pose_instability_detector.hpp"
+
+#include "tier4_autoware_utils/geometry/geometry.hpp"
+
+#include <tf2_geometry_msgs/tf2_geometry_msgs.hpp>
+
+#include <tf2/LinearMath/Quaternion.h>
+
+#include <string>
+
+PoseInstabilityDetector::PoseInstabilityDetector(const rclcpp::NodeOptions & options)
+: Node("pose_instability_detector", options),
+  threshold_linear_x_(this->declare_parameter<double>("threshold_linear_x")),
+  threshold_linear_y_(this->declare_parameter<double>("threshold_linear_y")),
+  threshold_linear_z_(this->declare_parameter<double>("threshold_linear_z")),
+  threshold_angular_x_(this->declare_parameter<double>("threshold_angular_x")),
+  threshold_angular_y_(this->declare_parameter<double>("threshold_angular_y")),
+  threshold_angular_z_(this->declare_parameter<double>("threshold_angular_z")),
+  set_first_odometry_(false)
+{
+  odometry_sub_ = this->create_subscription<Odometry>(
+    "~/input/odometry", 10,
+    std::bind(&PoseInstabilityDetector::callback_odometry, this, std::placeholders::_1));
+
+  twist_sub_ = this->create_subscription<TwistWithCovarianceStamped>(
+    "~/input/twist", 10,
+    std::bind(&PoseInstabilityDetector::callback_twist, this, std::placeholders::_1));
+
+  const double interval_sec = this->declare_parameter<double>("interval_sec");
+  timer_ = this->create_wall_timer(
+    std::chrono::duration<double>(interval_sec),
+    std::bind(&PoseInstabilityDetector::callback_timer, this));
```

Comment:
Is it intentional to use a wall_timer? 
Using a wall_timer might lead to unexpected intervals for verification, especially when playing rosbags or testing in AWSIM at a lower speed.

FYI: The following code is how to create a callback that follows `/clock` when `use_sim_time` is true and *wall_time* when false.
```cpp
  timer_ = rclcpp::create_timer(
    this, this->get_clock(), std::chrono::duration<double>(interval_sec),
    std::bind(&PoseInstabilityDetector::callback_timer, this));
```